src/apps/activity_assignments/domain/assignments.py

```diff
@@ -0,0 +1,46 @@
+from uuid import UUID
+
+from pydantic import BaseModel, root_validator
+
+from apps.activity_assignments.errors import (
+    ActivityAssignmentActivityOrFlowError,
+    ActivityAssignmentRespondentOrInvitationError,
+)
+from apps.shared.domain import InternalModel, PublicModel
+
+
+class ActivityAssignmentCreate(BaseModel):
+    activity_id: UUID | None
+    activity_flow_id: UUID | None
+    respondent_id: UUID | None
+    target_subject_id: UUID | None
+    invitation_id: UUID | None
+
+    @root_validator
+    def validate_assignments(cls, values):
+        if not values.get("activity_id") and not values.get("activity_flow_id"):
+            raise ActivityAssignmentActivityOrFlowError()
+
+        if not values.get("respondent_id") and not values.get("invitation_id"):
+            raise ActivityAssignmentRespondentOrInvitationError()
+
+        return values
+
+
+class ActivitiesAssignmentsCreate(InternalModel):
+    assignments: list[ActivityAssignmentCreate]
+
+
+class ActivityAssignment(PublicModel):
+    id: UUID
+    activity_flow_id: UUID | None
+    activity_id: UUID | None
+    flow_id: UUID | None
```

Comment:
Curious what the need for a redundant `flow_id` prop is here, when we already have `activity_flow_id` above? In my tests, it seems to always be returning `null` in endpoint responses, while `activity_flow_id` is correctly populated.

Reply:
my mystake, removed flow_id